src/visual.ts

```diff
@@ -465,25 +472,56 @@ export class DualKpi implements IVisual {
     }
 
     public update(options: VisualUpdateOptions) {
-        let dataView: DataView = this.dataView = options.dataViews && options.dataViews[0];
+        try {
+            const dataView: DataView = this.dataView = options.dataViews && options.dataViews[0];
 
-        if (!dataView ||
-            !dataView.metadata ||
-            !dataView.metadata.columns) {
+            if (!dataView ||
+                !dataView.metadata ||
+                !dataView.metadata.columns) {
 
-            this.displayRootElement(false);
+                this.displayRootElement(false);
 
-            return;
-        }
+                return;
+            }
+
+            this.formattingSettings = this.formattingSettingsService.populateFormattingSettingsModel(DualKpiSettingsModel, dataView);
+            this.formattingSettings.validateValues();
+            this.formattingSettings.setLocalizedOptions(this.localizationManager);
+            this.setHighContrastColors(this.colorHelper);
+
+            const data: IDualKpiData = this.data = DualKpi.converter(this.dataView, this.formattingSettings);
 
-        let data: IDualKpiData = this.data = DualKpi.converter(this.dataView, this.localizationManager, this.colorHelper);
+            const availableHeight = options.viewport.height < 90 ? 90 : options.viewport.height;
```

Comment:
lets create a constant for such values, like `const defaultVisualHeight = 90;` or similar. If we have many constant values might be used in other files, we can create a separate file.

Reply:
Done.